### PR TITLE
add save and load logic to fittable transform

### LINKED ISSUE #77: Add saving/loading logic for fittable transform
We will need a saving/loading logic for fittable transform so that we don't fit every time. It's like Pytorch's model checkpoint saving/loading.

### PR DESCRIPTION
**Pull request by yiweny**
Fixes #77

### CODE REVIEW

File: torch_frame/transforms/fittable_base_transform.py [hunk 1]
```diff
@@ -90,3 +98,11 @@ def _fit(self, tf: TensorFrame, col_stats: Dict[str, Dict[StatType, Any]]):
     @abstractmethod
     def _forward(self, tf: TensorFrame) -> TensorFrame:
         raise NotImplementedError
+
+    @abstractmethod
+    def _save(self, path: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def _load(self) -> None:
```
Comment (rusty1s) [inline torch_frame/transforms/fittable_base_transform.py:107]: This should be a `classmethod`.

File: torch_frame/transforms/fittable_base_transform.py [hunk 2]
```diff
@@ -90,3 +98,11 @@ def _fit(self, tf: TensorFrame, col_stats: Dict[str, Dict[StatType, Any]]):
     @abstractmethod
     def _forward(self, tf: TensorFrame) -> TensorFrame:
         raise NotImplementedError
+
+    @abstractmethod
+    def _save(self, path: str) -> None:
```
Comment (rusty1s) [inline torch_frame/transforms/fittable_base_transform.py:103]: Why does this need an underscore?

**Review by weihua916 — COMMENTED** [review]
Thanks for starting this!

Ideally, what we want is the following:

```python
transform = CatToNumTransform()
transform.fit(dataset.tensor_frame, dataset.col_stats)
# save the checkpoint
torch.save(transform.state_dict(), path)

# Initialize a completely new transform
loaded_transform = CatToNumTransform()
# Load from checkpoint
loaded_transform.load_state_dict(torch.load(path))
# Make sure loaded_transform and transform behaves the same
```

**Review by weihua916 — APPROVED** [review]
Looks good. Made some edits to simplify the test.

**Pull request merged**